src/proxy/outbound.rs

```diff
@@ -308,11 +310,15 @@ async fn connect_tls(
 ) -> Result<tokio_boring::SslStream<TcpStream>, tokio_boring::HandshakeError<TcpStream>> {
     connector.set_verify_hostname(false);
     connector.set_use_server_name_indication(false);
-    let addr = stream.local_addr();
-    connector.set_verify_callback(boring::ssl::SslVerifyMode::PEER, move |_, x509| {
-        info!("TLS callback for {:?}: {:?}", addr, x509.error());
-        true
-    });
+    // let addr = stream.local_addr();
```

Comment:
This seems like we are not verifying if its commented out; I think we need custom verification callback for SPIFFE possibly

Reply:
AFAICT calling set_verify_callback or set_verify multiple times is overriding and just using the value from the last call. The passed in `connector` has the verifier.